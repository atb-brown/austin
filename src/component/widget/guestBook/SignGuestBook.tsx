import { guestEntryBRadius, guestEntryHeight } from "./GuestEntry";
import React from "react";
import "../../../common.css";

/**
 * This component is a button that guests can click to sign my guest book.
 *
 * @return {SignGuestBook}
 */
export default function SignGuestBook(): JSX.Element {
  const message = "Sign my guest book!";
  return (
    <a
      className="clickable"
      href="https://github.com/atb-brown/guest-book/edit/main/guests.json"
      style={{
        alignItems: "center",
        alignSelf: "flex-end",
        borderBottomLeftRadius: guestEntryBRadius,
        borderBottomRightRadius: guestEntryBRadius,
        borderTopLeftRadius: guestEntryBRadius,
        color: "white",
        display: "flex",
        fontSize: "80%",
        height: guestEntryHeight,
        justifyContent: "center",
        textDecoration: "none",
        width: "15%",
      }}
    >
      {message}
    </a>
  );
}
