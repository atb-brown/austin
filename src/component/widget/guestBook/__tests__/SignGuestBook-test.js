import SignGuestBook from "../SignGuestBook";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

it("renders to match snapshot", () => {
  const tree = renderer.create(<SignGuestBook />);

  expect(tree).toMatchSnapshot();
});
