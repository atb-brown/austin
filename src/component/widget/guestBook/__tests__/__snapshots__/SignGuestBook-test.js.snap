// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<a
  href="https://github.com/atb-brown/guest-book/edit/main/guests.json"
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-end",
      "backgroundColor": "rebeccapurple",
      "borderBottomLeftRadius": 15,
      "borderBottomRightRadius": 15,
      "borderTopLeftRadius": 15,
      "color": "white",
      "display": "flex",
      "fontSize": "80%",
      "height": "85px",
      "justifyContent": "center",
      "textDecoration": "none",
      "width": "15%",
    }
  }
>
  Sign my guest book!
</a>
`;
